### PR TITLE
French Adjectives Transformation

### PR DESCRIPTION
**Pull request by Louanes1**

### CODE REVIEW

**Comment by Louanes1** [comment]
The pytest suceed when I try to run it locally, it shouldn't fail in the build : 
image

File: transformations/french_synonym_adjectives_transformation/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+## Supported Task
+
+This perturbation can be used for any French task.
+
+## What does it intend to benefit ?
+
+This perturbation would benefit all tasks which have a sentence/paragraph/document as input like text classification, text generation, etc. that requires synthetic data augmentation / diversification.
+
+## What are the limitation of this transformation ?
+This tool does not take the general context into account, sometimes, the ouput will not match the general sense of te sentence.
```
Comment (gxywang) [inline transformations/french_synonym_adjectives_transformation/README.md:20]: misspelling: *the

If possible, the results of the evaluate.py test should be added to the Readme as well.

File: transformations/french_synonym_adjectives_transformation/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Adjective Synonym Substitution 🦎  + ⌨️ → 🐍
+
+
+This transformation change some words with synonyms according to if their POS tag is a ADJ for simple french sentences. It requires Spacy_lefff (an extention of spacy for french POS and lemmatizing) and nltk package with the open multilingual wordnet dictionary.
+
+Authors : Lisa Barthe and Louanes Hamla from Fablab by Inetum in Paris
```
Comment (gxywang) [inline transformations/french_synonym_adjectives_transformation/README.md:6]: Emails should be added to the authors, please.

File: transformations/french_synonym_adjectives_transformation/test.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      "inputs": {
+        "sentence": "Nous sommes venu en grand nombre."
+      },
+      "outputs": [{
+        "sentence": "Nous sommes venu en important nombre."
+      }]
+
+    }
+
+
```
Comment (gxywang) [inline transformations/french_synonym_adjectives_transformation/test.json:38]: It is recommended to include at least 5 examples here.

File: transformations/french_synonym_adjectives_transformation/transformation.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import re
+from textblob_fr import PatternTagger, PatternAnalyzer
+import nltk
+nltk.download('wordnet')
+from textblob.wordnet import NOUN, VERB, ADV, ADJ
+import spacy
+from spacy_lefff import LefffLemmatizer, POSTagger
+from spacy.language import Language
+from nltk.corpus import wordnet
+import nltk
```
Comment (gxywang) [inline transformations/french_synonym_adjectives_transformation/transformation.py:11]: nltk is imported twice.

File: transformations/french_synonym_adjectives_transformation/transformation.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+
+class FrenchAdjectivesSynonymTransformation(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+        TaskType.TEXT_TAGGING,
+    ]
+    languages = ["fr"]
+
```
Comment (gxywang) [inline transformations/french_synonym_adjectives_transformation/transformation.py:74]: Relavent [keywords ](https://github.com/GEM-benchmark/NL-Augmenter/blob/main/docs/keywords.md) should also be added.

File: transformations/french_synonym_adjectives_transformation/requirements.txt [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#fr core news md model :
+fr-core-news-md @ https://github.com/explosion/spacy-models/releases/download/fr_core_news_md-3.0.0/fr_core_news_md-3.0.0-py3-none-any.whl
+spacy-lefff==0.4.0
+textblob_fr==0.2.0
+nltk
```
Comment (gxywang) [inline transformations/french_synonym_adjectives_transformation/requirements.txt:5]: Should nltk's version be specified as well?

File: transformations/french_synonym_adjectives_transformation/transformation.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    def __init__(self, seed=0, max_outputs=1):
+        super().__init__(seed, max_outputs=max_outputs)
+
+    def generate(self, sentence : str):
+        perturbed_texts = synonym_transformation(
+            sentence
+        )
+        print("perturbed text inside of class",perturbed_texts)
```
Comment (gxywang) [inline transformations/french_synonym_adjectives_transformation/transformation.py:82]: May want to delete this print line.

**Review by gxywang — CHANGES_REQUESTED** [review]
Hi @Louanes1, I'm one of the assigned reviewers for your transformation. Below is my general review, see comments for more specific feedback.

Correctness: Failed the build with an AssertionError. I saw the comment regarding passing the unit test locally. Maybe see if your local version is up-to-date? Otherwise, I'm also unsure why it failed.
Interface: Looks good.
Applicable Tasks & Keywords: Task types have been added, missing keywords.
Specificity: The task is specific to text classification, generation, and tagging.
Novelty: This transformation is not implemented in NL-Augmenter already.
Adding New Libraries: All added to requirements, only nltk's version has not been specified.
Description: Readme looks good.
Data and code source: None. Should add a sentence stating it was fully created by author (or cite relevant sources).
Paraphrasers and Data Augmenters: N/A
Test Cases: 3 examples added, 5 recommended.
Evaluating Robustness: Evaluation has not been provided.
Languages other than English: French.
Decent Programming Practise: No docstrings, consider adding some.

If anything needs clarification, please let me know. Otherwise, thanks for the contribution!

**Comment by mille-s** [comment]
Should there be just one transformation for Verbs https://github.com/GEM-benchmark/NL-Augmenter/pull/250, Nouns https://github.com/GEM-benchmark/NL-Augmenter/pull/247 and Adjectives, since the method seems to be the same? The transformation could be called "Lexical susbstitutions for French".

**Comment by kaustubhdhole** [comment]
Agree with @mille-s  above. By default, you can return all (nouns,adjectives and adverbs) and still have it parametrized. Since we are tabulating the PRs, I am closing this. Please try to address all the suggested changes of the reviews in all the 3 PRs to https://github.com/GEM-benchmark/NL-Augmenter/pull/250

**Pull request closed without merging**